### ISSUE TITLE
Rust Analyzer misidentifying None in match as a variable

### ISSUE THREAD

**Issue by jeramyRR** [rust-lang/rust-analyzer#20075]
The Rust Analyzer VSCode extension is flagging `None` in a match as a variable and complaining that it's not in snake_case.  I don't get any warnings from `cargo clippy` or `cargo check`, just rust analyzer.

**rust-analyzer version**: 0.3.2509

**rustc version**:  1.87.0 (17067e9ac 2025-05-09)

**editor or extension**:  VSCode 0.3.2509

**code snippet to reproduce**:
```rust
let opt = Some(true);
match opt {
    Some(_) => {
        tracing::info!(
            "Logging initialized with level: {log_level}, name: {log_name}, format: {log_format:?}"
        );
    }
    None => {  #<------ Variable `None` should have snake_case name, e.g. `none`rust-analyzer[non_snake_case]...
        tracing::warn!("Logging initialization failed, using default settings.");
    }
}

```

**example code screenshot**:
<img width="464" alt="Image" src="https://github.com/user-attachments/assets/1450453c-c449-4236-b35a-52ffcb76e1c1" />

**Comment by ChayimFriedman2** [comment]
That's usually a sign of a corrupted sysroot. First: does goto def on `None` works? Second: try `rustup component remove rust-src` then `rustup component install rust-src`.

**Comment by jeramyRR** [comment]
Thank you @ChayimFriedman2.  Removing rust-src and re-installing it fixed that warning.
